service/controller/resource/s3object/desired_test.go

```diff
@@ -6,41 +6,87 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	releasev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned/fake"
 	"github.com/giantswarm/certs/certstest"
 	"github.com/giantswarm/micrologger/microloggertest"
 	"github.com/giantswarm/randomkeys/randomkeystest"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/aws-operator/pkg/label"
 	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
 )
 
 func Test_DesiredState(t *testing.T) {
 	t.Parallel()
-	clusterTpo := &v1alpha1.AWSConfig{
-		Spec: v1alpha1.AWSConfigSpec{
-			Cluster: v1alpha1.Cluster{
+	clusterTpo := &providerv1alpha1.AWSConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				label.ReleaseVersion: "1.0.0",
+			},
+		},
+		Spec: providerv1alpha1.AWSConfigSpec{
+			Cluster: providerv1alpha1.Cluster{
 				ID: "test-cluster",
 			},
 		},
 	}
 
+	release := releasev1alpha1.NewReleaseCR()
+	release.ObjectMeta.Name = "v1.0.0"
+	release.Spec.Components = []releasev1alpha1.ReleaseSpecComponent{
+		{
+			Name:    "kubernetes",
+			Version: "1.15.4",
+		},
+		{
+			Name:    "calico",
+			Version: "3.9.1",
+		},
+		{
+			Name:    "etcd",
+			Version: "3.3.15",
+		},
+	}
+	clientset := fake.NewSimpleClientset(release)
+
 	masterKeyPattern := "cloudconfig/v[\\d_]+/master"
 	workerKeyPattern := "cloudconfig/v[\\d_]+/worker"
 
 	masterKeyRegexp := regexp.MustCompile(masterKeyPattern)
 	workerKeyRegexp := regexp.MustCompile(workerKeyPattern)
 
 	testCases := []struct {
-		obj            *v1alpha1.AWSConfig
+		obj            *providerv1alpha1.AWSConfig
 		description    string
 		expectedBucket string
 		expectedBody   string
+		template       string
 	}{
 		{
-			description:    "basic match",
+			description:    "case 0: basic match",
 			obj:            clusterTpo,
 			expectedBody:   "mybody-",
+			template:       "mybody-",
+			expectedBucket: "myaccountid-g8s-test-cluster",
+		},
+		{
+			description: "case 1: template hyperkube",
```

Comment:
The thing being tested here isn't really related to `case 0`, but it still feels like the most reasonable place to put it.